aria_data_loaders/aria_data_utils/aria_live_streamer.py

```diff
@@ -0,0 +1,598 @@
+import os
+from typing import Any, Dict, List, Optional, Tuple
+
+import aria.sdk as aria
+import click
+import cv2
+import matplotlib.pyplot as plt
+import numpy as np
+import projectaria_tools.core as aria_core
+import rospy
+import sophus as sp
+from aria_data_utils.aria_sdk_utils import update_iptables
+from aria_data_utils.conversions import (
+    np_matrix3x4_to_sophus_SE3,
+    ros_Pose_to_sophus_SE3,
+    sophus_SE3_to_ros_Pose,
+    sophus_SE3_to_ros_TransformStamped,
+)
+from aria_data_utils.detector_wrappers.april_tag_detector import AprilTagDetectorWrapper
+from aria_data_utils.detector_wrappers.object_detector import ObjectDetectorWrapper
+from aria_data_utils.math_utils import get_running_avg_world_T_b
+from cairo import Device
+from geometry_msgs.msg import Pose, PoseStamped
+from nav_msgs.msg import Odometry
+from projectaria_tools.core.calibration import (
+    device_calibration_from_json_string,
+    distort_by_calibration,
+    get_linear_camera_calibration,
+)
+from projectaria_tools.core.sensor_data import ImageDataRecord
+from scipy.spatial.transform import Rotation
+from spot_rl.utils.utils import ros_frames as rf
+from tf2_ros import StaticTransformBroadcaster
+from visualization_msgs.msg import Marker
+
+# For information on useful reference frame, please check https://github.com/facebookresearch/spot-sim2real?tab=readme-ov-file#eyeglasses-run-spot-aria-project-code
+
+
+class AriaLiveReader:
+    """
+    Class to livestream frames and publish different data on ROS topics
+
+    This class reads images from Aria's camera and runs process_frame method on each frame,
+    which has following detectors:
+        1. AprilTagDetectorWrapper: Detects QR codes in the image.
+        2. ObjectDetectorWrapper: Detects objects in the image
+
+    QR Detector
+    We get a device_T_marker for each frame in which qr code is detected.
+    Depending on the frame rate of image capture, multiple frames may have captured the qr code.
+    Averaging all transforms would be best way to compensate for any noise that may exist in any frame's detections
+    camera_T_marker is used to compute device_T_marker[i] and thus ariaWorld_T_marker[i].
+    Then we average all ariaWorld_T-marker to find average marker pose wrt ariaWorld.
+
+    Object Detector
+    We use multi-class object detection models to be able to detect multiple objects in an image.
+    Whem we detect an object, we store aria's pose (in cpf frame w.r.t ariaWorld) at the location.
+
+
+    Args:
+        verbose (bool): If true, prints debug messages
+
+    Publishers:
+        /aria_current_pose: PoseStamped msg containing current pose of aria (cpf) w.r.t ariaWorld
+        /aria_pose_of_interest: PoseStamped msg containing pose of interest (cpf) w.r.t ariaWorld
+        /aria_odom: Odometry msg containing odometry info (in cpf) w.r.t ariaWorld
+        /aria_pose_of_interest_marker: Marker msg containing pose of interest (cpf) w.r.t ariaWorld
```

Comment:
not sure what this comment means. Is it the pose of Aria when the marker was seen?

Reply:
This topic publishes a Marker message so we may visualize `/aria_pose_of_interest` in rviz. We went with a ROS marker as we could make it whatever shape+color we wanted and disambiguate w/`/aria_current_pose` easily via code for debugging.